src/time.rs

```diff
@@ -587,43 +597,103 @@ impl TryFrom<&str> for MicrosecondsPrecisionOverflowBehavior {
     }
 }
 
+/// Defines how timestamps should be interpreted
+#[derive(Debug, Clone, Default, Copy, PartialEq)]
+pub enum TimestampInterpretation {
+    /// Automatically determine if the timestamp is in seconds or milliseconds
+    /// based on its value (default behavior)
+    #[default]
+    Auto,
+    /// Always interpret timestamps as seconds, regardless of their value
+    AlwaysSeconds,
+}
+
+impl TryFrom<&str> for TimestampInterpretation {
+    type Error = ConfigError;
+    fn try_from(value: &str) -> Result<Self, ConfigError> {
+        match value.to_lowercase().as_str() {
+            "auto" => Ok(Self::Auto),
+            "always_seconds" => Ok(Self::AlwaysSeconds),
+            _ => Err(ConfigError::UnknownTimestampInterpretationString),
+        }
+    }
+}
+
+/// Configuration options for time parsing and handling
 #[derive(Debug, Clone, Default, PartialEq)]
 pub struct TimeConfig {
+    /// Defines how to handle microsecond precision overflow
     pub microseconds_precision_overflow_behavior: MicrosecondsPrecisionOverflowBehavior,
+    /// Optional timezone offset for Unix timestamps
     pub unix_timestamp_offset: Option<i32>,
+    /// Defines how timestamps should be interpreted
+    pub timestamp_interpretation: TimestampInterpretation,
```

Comment:
It feels to me like this is actually config for `DateTime`, not `Time`.

So I suggest we make a `DatetimeConfig { time_config: TimeConfig, timestamp_interpretation: TimestampInterpretation }` in `datetime.rs`

It might also be worth trying out this branch against `pydantic-core` to see how painful it is to adjust to needing a different `DatetimeConfig` type.

Reply:
Thank you for the suggestion, @davidhewitt. I'll implement the `DatetimeConfig` in `datetime.rs` and aim to ensure backward compatibility. I'll test this branch against `pydantic-core` and update you on the results.